cmd/scollector/collectors/database.go

```diff
@@ -0,0 +1,345 @@
+package collectors
+
+import (
+	"database/sql"
+	"fmt"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	_ "bosun.org/_third_party/github.com/go-sql-driver/mysql"
+	"bosun.org/cmd/scollector/conf"
+	"bosun.org/collect"
+	"bosun.org/metadata"
+	"bosun.org/opentsdb"
+	"bosun.org/slog"
+)
+
+type DatabaseQuery struct {
+	name        string
+	query       string
+	description string
+	hasTime     bool
+	interval    time.Duration
+	next        time.Time
+}
+
+var DatabaseVars struct {
+	vaildDatabases map[string]bool
+}
+
+func init() {
+	DatabaseVars.vaildDatabases = map[string]bool{"mysql": true}
+}
+
+func Database(c conf.Database) error {
+	return DatabaseAddCollector(c)
+}
+
+func DatabaseAddCollector(c conf.Database) error {
+	_, found := DatabaseVars.vaildDatabases[c.Type]
+	if !found {
+		return fmt.Errorf("%v: %v", "invalid Database Type", c.Type)
+	}
+	if c.DBName != "" {
+		cleaned, _ := opentsdb.Clean(c.DBName)
```

Comment:
Opentsdb's allowed characters probably are not the same as valid names for mysql databases. What are the rules for mysql?